### ISSUE TITLE
NPE while accessing a repository requiring authentication

### ISSUE THREAD

**Issue by cpotter1** [mguymon/naether#16]
I added a repository that requires authentication and I noticed two issues.

First, since maven central does not have this particular artifact, you get a NPE.  I then decided to reorder the remote repository list (since Naether is using a LinkedHashSet) and got a little further.  Here is the code I used to reverse order the repos:
```java
    // hack to force central to be last
    Set<RemoteRepository> remotes = instance.getRemoteRepositories();
    List<RemoteRepository> remotesList = new ArrayList<RemoteRepository>(remotes);
    Collections.reverse(remotesList);
    instance.setRemoteRepositories(new LinkedHashSet<RemoteRepository>(remotesList));
```

After this I became blocked on the issue with the NPE related to authentication.  Seems like its happening in the Aether / wagon code though.

Caused by: org.sonatype.aether.transfer.ArtifactTransferException: Could not transfer artifact com.xyz.mmx:mmx-client-java:pom:2.3.98_D1_RC1 from/to dev-repo.xyz.com-artifactory-everything--8081 (http://dev-repo.xyz.com:8081/artifactory/everything/): NullPointerException
	at org.sonatype.aether.connector.wagon.WagonRepositoryConnector$4.wrap(WagonRepositoryConnector.java:951)
	at org.sonatype.aether.connector.wagon.WagonRepositoryConnector$4.wrap(WagonRepositoryConnector.java:941)
	at org.sonatype.aether.connector.wagon.WagonRepositoryConnector$GetTask.run(WagonRepositoryConnector.java:669)
	at org.sonatype.aether.util.concurrency.RunnableErrorForwarder$1.run(RunnableErrorForwarder.java:60)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)
Caused by: java.lang.NullPointerException: null
	at org.apache.maven.wagon.providers.http.LightweightHttpWagon.openConnectionInternal(LightweightHttpWagon.java:276)
	at org.apache.maven.wagon.AbstractWagon.openConnection(AbstractWagon.java:115)
	at org.apache.maven.wagon.AbstractWagon.connect(AbstractWagon.java:217)
	at org.sonatype.aether.connector.wagon.WagonRepositoryConnector.connectWagon(WagonRepositoryConnector.java:345)
	at org.sonatype.aether.connector.wagon.WagonRepositoryConnector.pollWagon(WagonRepositoryConnector.java:385)
	at org.sonatype.aether.connector.wagon.WagonRepositoryConnector$GetTask.run(WagonRepositoryConnector.java:571)
	at org.sonatype.aether.util.concurrency.RunnableErrorForwarder$1.run(RunnableErrorForwarder.java:60)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)

**Comment by cpotter1** [comment]
Yeah, the problem is definitely with the latest version of Aether, not the version you are using.  I wish those guys would use enforcer plugin because there are other issues they have.

**Comment by cpotter1** [comment]
If you don't mind, please also deploy the latest version.  The source shows 0.13.7, but the official deployment at https://oss.sonatype.org/content/groups/public/com/tobedevoured/naether/core shows 0.13.5

**Comment by mguymon** [comment]
I can push that out laster on today, I am having problems push a new version to maven central.

**Comment by mguymon** [comment]
released 0.13.7
